libensemble/executors/mpi_executor.py

```diff
@@ -157,11 +159,14 @@ def _launch_with_retries(self, task: Task, subgroup_launch: bool, wait_on_start:
                     )
             except Exception as e:
                 logger.warning(f"task {task.name} submit command failed on try {retry_count} with error {e}")
+                task.state = "FAILED_TO_START"
```

Comment:
Looking at how `Task` is implemented, I think there might be a risk that `Task._set_complete` will override this state. Do you think this can be an issue?